pipeline/routing/ppr.py

```diff
@@ -165,23 +166,25 @@ def createPathNetwork(cur, path, edges, relation_id, dhid_from, dhid_to):
     
     for edge in edgeIter:
         if requiresAccessSpace(previousEdge, edge): # checks whether the given parameters need the creation of an access space
-            newDHID = insertAccessSpaces(cur, edge, previousEdge, relation_id) # returns a newly created DHID for the access space
-            pathId = insertPathLink(cur, relation_id, pathLink, previousDHID, newDHID)
+            newDHID, toLevel = insertAccessSpaces(cur, edge, previousEdge, relation_id) # returns a newly created DHID for the access space and the level of the access space
+            pathId = insertPathLink(cur, relation_id, pathLink, previousDHID, newDHID, abs(fromLevel - toLevel))
             if pathId:
                 insertPathsElementsRef(cur, pathId, pathLinkEdges)
             pathLink = edge["path"] # create a new pathLink consisting of the current edge
             pathLinkEdges = [edge]
             previousDHID = newDHID
+            fromLevel = toLevel
         else:
             # append all but the first node of the edge, because the first node is the same as the last node of the previous edge
             # use extend, because there can be multiple nodes in the edge (polyline)
             pathLink.extend(edge["path"][1:])
             pathLinkEdges.append(edge)
+            toLevel = edge["level"]
 
         previousEdge = edge
     
     # the last part of the path is not inserted yet (between the last access space and the stop_area_element 'dhid_to')
-    pathId = insertPathLink(cur, relation_id, pathLink, previousDHID, dhid_to)
+    pathId = insertPathLink(cur, relation_id, pathLink, previousDHID, dhid_to, abs(fromLevel - toLevel))
```

Comment:
As a bonus info for the future, we could keep the sign to indicate the "incline". In other words whether the path link goes upwards or downwards.

Should then be added to the existing code documentation in the `02_setup.sql` file.

I guess the `abs()` method then needs to be moved to the estimate duration function.

Reply:
Added in [794b496](https://github.com/OPENER-next/osm2vdv462/commit/794b49679eb365ffa1f3e6b07df9c4b2785d92fd)